mri2mrs.py

```diff
@@ -0,0 +1,218 @@
+            print("Invalid label map format, exiting ...")
+
+        # if output directory exists, delete and re-create
+        if os.path.exists(self.outDir):
+            rm.run(['-r', self.outDir])
+
+        os.makedirs(self.outDir)
+
+
+        scriptDir= os.getcwd()
+        os.chdir(self.outDir)
+
+        global f, logFile
+        logFile= 'log-{}.txt'.format(self.case)
+        f= open(logFile, 'w')
+
+        # fsl requires nifti
+        if self.img.endswith('.nrrd') or self.img.endswith('.nhdr'):
+            convertedImg= self.case+'-t1w.nii.gz'
+            ((ConvertBetweenFileFormats[self.img, convertedImg] ) >> logFile)()
+
+        else:
+            convertedImg= self.img
+
+
+
+        if self.mask:
+            log('Creating mask and multiplying the input image by mask ...')
+            bet_mask= self.case+'-t1w-mask.nii.gz'
+            ((bet[convertedImg, bet_mask, '-m -n -f', self.betThreshold] ) >> logFile)()
+
+            processedImg = self.case + '-t1w-bet.nii.gz'
+            fslmaths[convertedImg, '-mul', bet_mask, processedImg] & FG
+
+        else:
+            processedImg= convertedImg
+
+        warpDim(processedImg)
+
+        log('Segmenting T1 image to white/gray/CSF ...')
+        ((fast['-o', 'fast_out', processedImg] ) >> logFile)()
+
+
+        log('Creating white, gray, csf separate images ...')
+
+        white = 'fast_out_seg.nii.gz'
+        gray = 'fast_out_seg2.nii.gz'
+        csf = 'fast_out_seg3.nii.gz'
+
+
+        fslmaths[white, '-uthr', '1', gray] & FG
+        fslmaths[gray, '-add', '1', gray] & FG
+        fslmaths[gray, '-uthr', '1', gray] & FG
+
+        fslmaths[white, '-thr', '3', csf] & FG
+        fslmaths[csf, '-div', '3', csf] & FG
+
+
+        # Ofer's MATLAB code reads only .nhdr file
+        compatibleImg= self.case+'-t1w.nhdr'
+        # TODO: Check with Ofer if he wants unprocessed image as the first argument
+        ((ConvertBetweenFileFormats[processedImg, compatibleImg] ) >> logFile)()
+
+        try:
+            regionPrefix = self.region
+            log('Defining MRS on T1 image using MATLAB ...')
+
+            retcode, stdout, stderr= matlab.run(['-singleCompThread', '-nojvm', '-nosplash', '-r',
+                   "addpath {}, MRStoAnatomy(\'{}\', \'{}\', \'{}\', \'{}_mask\'), exit"
+                   .format(scriptDir, 'tmp-sb.nhdr', compatibleImg, self.labelMap, regionPrefix)])
+            externalOutput(retcode, stdout, stderr)
+
+
+            log('Combining the mask with the zero files to create a mask having same dimensions as that of T1, '
+                'launching Slicer ...')
+
+            retcode, stdout, stderr= Slicer.run(['--launch', 'AddScalarVolumes',
+                   self.region+'_mask_zr.nhdr',
+                   self.region + '_mask_jm.nhdr',
+                   '--order', '0',
+                   self.region + '_mask.nhdr'])
+            externalOutput(retcode, stdout, stderr)
+
+
+        except:
```

Comment:
If you are going to catch the exception then please print it out so someone knows what happened:

```
       except Exception as e:
            log('MRS mask creation failed for {}, exiting due to exception: \n{} ...'.format(self.img, e))
```

Reply:
This aspect of the design has been improved. Now, the command, stdout, stderr are printed both in the console and the output-log-file.